### PR TITLE
[it] community-contribute-code.md

### PR DESCRIPTION
**Pull request by codeadmin-peritiae**
Basic translation

### CODE REVIEW

File: site/it/community/contribute/code.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# Contributo al codice TensorFlow
```
Comment (bhack) [inline site/it/community/contribute/code.md:1]: Contribuire

File: site/it/community/contribute/code.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+Sia che tu voglia aggiungere una funzione perdita, migliorare la copertura di un test, oppure scrivere una
```
Comment (bhack) [inline site/it/community/contribute/code.md:3]: I don't think loss is translated

Comment (bhack) [reply site/it/community/contribute/code.md:3]: Also check https://accademiadellacrusca.it/it/consulenza/genere-dei-forestierismi/1229 about cases like this one. Is it more common `l'RFC/un RFC` or `la RFC/una RFC`?

File: site/it/community/contribute/code.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+RFC per un cambiamento rilevante di progettazione, questa parte della guida del contributore ti sarà d'aiuto
+per cominciare. Grazie per il lavoro e l'interesse nel miglioramento di TensorFlow.
+
+## Prima di cominciare
+
+Prima di contribuire al codice sorgente di un progetto TensorFlow, cortesemente consulta il file `CONTRIBUTING.md` nel repository GitHub del progetto. (Per esempio, vedi
+[file CONTRIBUTING.md nel repository centrale di TensorFlow](https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md).) A tutti i contributori è richiesto di firmare un [Accordo di Licenza di Contribuzione](https://cla.developers.google.com/clas) (CLA).
+
+Per evitare attività duplicate, prima di cominciare a lavorare su una funzionalità non banale, cortesemente consulta [RFCs correnti](https://github.com/tensorflow/community/tree/master/rfcs) e contatta gli sviluppatori nei forum TensorFlow. Talvolta siamo selettivi quando dobbiamo decidere se aggiungere una funzionalità, ed il modo migliore di contribuire ed aiutare il progetto è lavorare su problemi conosciuti.
```
Comment (bhack) [inline site/it/community/contribute/code.md:12]: `gli/le RFCs` see the previous comment

File: site/it/community/contribute/code.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+per cominciare. Grazie per il lavoro e l'interesse nel miglioramento di TensorFlow.
+
+## Prima di cominciare
+
+Prima di contribuire al codice sorgente di un progetto TensorFlow, cortesemente consulta il file `CONTRIBUTING.md` nel repository GitHub del progetto. (Per esempio, vedi
+[file CONTRIBUTING.md nel repository centrale di TensorFlow](https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md).) A tutti i contributori è richiesto di firmare un [Accordo di Licenza di Contribuzione](https://cla.developers.google.com/clas) (CLA).
+
+Per evitare attività duplicate, prima di cominciare a lavorare su una funzionalità non banale, cortesemente consulta [RFCs correnti](https://github.com/tensorflow/community/tree/master/rfcs) e contatta gli sviluppatori nei forum TensorFlow. Talvolta siamo selettivi quando dobbiamo decidere se aggiungere una funzionalità, ed il modo migliore di contribuire ed aiutare il progetto è lavorare su problemi conosciuti. 
+
+## Temi per il nuovo contributore
```
Comment (bhack) [inline site/it/community/contribute/code.md:14]: `Temi` could be untraslated `Issues` cause is referring to Github Issue that is not translated

File: site/it/community/contribute/code.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+I nuovi contributori alla ricerca di un primo tema per un contributo al codice TensorFlow, dovrebbero cercare i seguenti tag. Raccomandiamo fortemente che i nuovi contributori affrontino inizialmente progetti “easy” e problematiche "good first issue"; questo aiuta il contributore a familiarizzarsi con il flusso di lavoro della contribuzione, ed agli sviluppatori del core di conoscere il contributore.
```
Comment (bhack) [inline site/it/community/contribute/code.md:16]: Same here for `tema`

Comment (bhack) [reply site/it/community/contribute/code.md:16]: `familiarizzare`

Comment (bhack) [reply site/it/community/contribute/code.md:16]: `di contribuzione`

File: site/it/community/contribute/code.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+- `good first issue`
+- `easy`
+- `contributions welcome`
+
+Se siete interessati a reclutare un team per aiutarvi ad affrontare un problema ampio o una nuova funzionalità, cortemente mandate una email a [gruppo sviluppatori@](https://groups.google.com/a/tensorflow.org/forum/#!forum/developers) e osservate la nostra lista attuale di RFC.
```
Comment (bhack) [inline site/it/community/contribute/code.md:22]: "cortesemente" but in Italian we could skip the translation of please in many cases. See the old good http://tp.linux.it/buona_traduzione.html

File: site/it/community/contribute/code.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+- `good first issue`
+- `easy`
+- `contributions welcome`
+
+Se siete interessati a reclutare un team per aiutarvi ad affrontare un problema ampio o una nuova funzionalità, cortemente mandate una email a [gruppo sviluppatori@](https://groups.google.com/a/tensorflow.org/forum/#!forum/developers) e osservate la nostra lista attuale di RFC. 
+
+
+## Revisione del codice
+
+Nuove funzioni, eliminazione di difetti, ed ogni altro cambiamento nel codice base sono soggetti a revisione del codice.
```
Comment (bhack) [inline site/it/community/contribute/code.md:27]: I think we could keep bugs

File: site/it/community/contribute/code.md (hunk 5 above)
Comment (bhack) [inline site/it/community/contribute/code.md:16]: `codice base` you have used `codice base` later

File: site/it/community/contribute/code.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+La revisione del codice ricevuto come contributo in un progetto attraverso le "pull request" è una componente fondamentale dello sviluppo di TensorFlow. Incoraggiamo chiunque a cominciare rivedendo codice sottoposto da altri sviluppatori, specialmente in quei casi in cui il codice implementa qualcosa che ti piacerebbe usare.
```
Comment (bhack) [inline site/it/community/contribute/code.md:29]: `rivedendo` -> `revisionando`

File: site/it/community/contribute/code.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+Qui alcune domande da tenere a mente durante il processo di revisione del codice:
+
+*   *Vogliamo questa cosa in TensorFlow?* Potrebbe piacere usarla? Come utente TensorFlow, ti piacerebbe questa modifica e la useresti? Si tratta di un cambiamento nell'ambito di TensorFlow? Il costo di manutenzione di questa nuova funzione sarebbe maggiore dei suoi benefici?
+*   *Il codice è consistente con le API di TensorFlow?* Le funzioni pubbliche, le classi, ed i parametri sono nominati in modo appropriato e progettati intuitivamente?
+*   *E' inclusa la documentazione?* Le funzioni pubbliche, le classi, i tipi di ritorno e gli attributi memorizzati sono nominati in accordo con le convenzioni di TensorFlow e sono documentati chiaramente? La nuova funzionalità è descritta nella documentazione di TensorFlow ed illustrata, ovunque possibile, con esempi? La documentazione si legge in modo appropriato?
+
+*   *Il codice è leggibile da una persona?* Si evitano ridondanze? I nomi delle variabili dovrebbero essere migliorati per aumentare la chiarezza o la consistenza? Dovrebbero essere aggiunti commenti? Ci sono commenti inutili o estranei che potrebbero essere rimossi?
```
Comment (bhack) [inline site/it/community/contribute/code.md:37]: we could use also `è facilmente leggibile`

File: site/it/community/contribute/code.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+*   *Il codice è efficiente?* Potrebbe essere riscritto facilmente per essere più efficiente?
+*   IL codice è *retro-compatibile* con le versioni precedenti di TensorFlow?
+*   Il nuovo codice aggiungerà *nuove dipendenze* on other libraries?
+
+## Test e miglioramento della copertura dei test
+
+Unit testing di alta qualità è una pietra miliare del processo di sviluppo di TensorFlow. Per questo scopo usiamo immagini Docker. Le funzioni di test hanno nomi appropriati, e sono responsabili del controllo della validità degli algoritmi, così come delle varie opzioni del codice.
```
Comment (bhack) [inline site/it/community/contribute/code.md:44]: `Lo` or `Un`

File: site/it/community/contribute/code.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+Tutt le nuove funzioni e le correzioni di difetti *devono* includere test con adeguata copertura. Perciò accogliamo anche contributi di nuovi casi di test o migliorie ai test esistenti. Se scoprite che i nostri attuali test non sono completi — anche se ciò, al momento, non causa un difetto — per cotesia, sottoponete un issue e, se possibile una pull request.
```
Comment (bhack) [inline site/it/community/contribute/code.md:46]: `Tutte` and we could untraslate `bugs`

File: site/it/community/contribute/code.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+Per tutti i dettagli particolari sulle procedure di test in ciascun progetto TensorFlow, vedere i file `README.md` e `CONTRIBUTING.md` nel rispettivo repository su GitHub.
+
+Di particolare importanza è fare *testing adeguato*:
+
+*   E stata testata *ogni funzione pubblica di ogni classe*? 
+*   Sono stati testati *insiemi ragionevoli di parametri*, di valori, tipi e in ragionevoli combinazioni? 
+*   I test sono in grado di validare che *il codice è corretto*, e che esso faccia *ciò che la documentazione dice* che esso faccia?
+*   Se la modifica è la correzione di un difetto, è incluso un *test di non regressione*?
+*   Il test *supera il build in continuous integration*?
+*   Il test *copre tutte le linee del codice?* se no, le eccezioni sono ragionevoli ed esplicite?
```
Comment (bhack) [inline site/it/community/contribute/code.md:57]: Se

File: site/it/community/contribute/code.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+
+Se individuate dei problemi, cortesemente, considerate la possibilità di supportare un contributore a capire questi problemi ed a risolverli.
```
Comment (bhack) [inline site/it/community/contribute/code.md:59]: We could remove `cortesemente`.

File: site/it/community/contribute/code.md [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+*   Se la modifica è la correzione di un difetto, è incluso un *test di non regressione*?
+*   Il test *supera il build in continuous integration*?
+*   Il test *copre tutte le linee del codice?* se no, le eccezioni sono ragionevoli ed esplicite?
+
+Se individuate dei problemi, cortesemente, considerate la possibilità di supportare un contributore a capire questi problemi ed a risolverli. 
+
+
+## Miglioramento dei messaggi di errore o dei log
+
+Accogliamo contributi che migliorano la messaggistica di errore e la tracciatura.
```
Comment (bhack) [inline site/it/community/contribute/code.md:64]: We need to take a decision about logging if it is an untranslated word.

File: site/it/community/contribute/code.md [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+
+
+## Miglioramento dei messaggi di errore o dei log
+
+Accogliamo contributi che migliorano la messaggistica di errore e la tracciatura. 
+
+
+## Flusso dei contributi
+
+Tutti i contributi al codice—correzioni, nuovi sviluppi, miglioramenti ai test—seguono un flusso centrato su GitHub. Per partecipare allo sviluppo in TensorFlow, aprite un account GitHub. Quindi:
```
Comment (bhack) [inline site/it/community/contribute/code.md:69]: incentrato

File: site/it/community/contribute/code.md [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+
+    `$ git add -A`
+
+    `$ git commit -m "commit message here"`
+
+6.  Fate un push delle vostre modifiche sul vostro repository GitHub.
+
+    `$ git push origin branch-name`
+
+7.  Aprite una *Pull Request* (PR). Andando su repository originale del progetto su GitHub. Ci sarà un'evidenza a proposito del branch di cui avete fatto il push di recente, che vi chiederà se avete l'intenzione di aprire una pull request. Scegliete, *confronto tra repository* (*compare across repositories* n.d.t), e sottoponete la PR. Ciò produrrà l'invio di una email ai committer. Per dare maggiore visibilità potete considerare la possibilità di inviare anche un'email alla mailing list. (Per maggiori dettagli, vedere la [guida di GitHub sulle PR](https://help.github.com/articles/creating-a-pull-request-from-a-fork).
```
Comment (bhack) [inline site/it/community/contribute/code.md:96]: `una segnalazione in evidenza relativa al branch`

File: site/it/community/contribute/code.md [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+
+    `$ git commit -m "commit message here"`
+
+6.  Fate un push delle vostre modifiche sul vostro repository GitHub.
+
+    `$ git push origin branch-name`
+
+7.  Aprite una *Pull Request* (PR). Andando su repository originale del progetto su GitHub. Ci sarà un'evidenza a proposito del branch di cui avete fatto il push di recente, che vi chiederà se avete l'intenzione di aprire una pull request. Scegliete, *confronto tra repository* (*compare across repositories* n.d.t), e sottoponete la PR. Ciò produrrà l'invio di una email ai committer. Per dare maggiore visibilità potete considerare la possibilità di inviare anche un'email alla mailing list. (Per maggiori dettagli, vedere la [guida di GitHub sulle PR](https://help.github.com/articles/creating-a-pull-request-from-a-fork). 
+
+8.  I maintainer ed altri contributori *rivedranno la vostra PR*. Cortesemente partecipate alla conversazione, e provate ad *eseguire ogni richiesta di cambiamento*. Una volta che la PR è approvata, il codice sarà incorporato.
```
Comment (bhack) [inline site/it/community/contribute/code.md:98]: revisioneranno. Remove cortesemente

File: site/it/community/contribute/code.md [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+
+*Prima di mettervi a lavorare sul vostro prossimo contributo*, assicuratevi che il vostro repository locale sia aggiornato.
+
+1. Impostate il riferimento remoto. (Basta farlo una sola volta per ogni progetto, non ogni volta.)
```
Comment (bhack) [inline site/it/community/contribute/code.md:102]: Untranslate `upstream`

**Comment by bhack** [comment]
@lamberta Can we have some kind of automation to prompt the change side by side witih new translated file and the original version (at least a link to the original mapped file so that we can open quickly in a tab). This could really speedup the review.

File: site/it/community/contribute/code.md (hunk 2 above)
Comment (codeadmin-peritiae) [reply site/it/community/contribute/code.md:3]: > I don't think loss is translated

loss function in Italian is "funzione obiettivo": https://it.wikipedia.org/wiki/Funzione_obiettivo

Comment (codeadmin-peritiae) [reply site/it/community/contribute/code.md:3]: > Also check https://accademiadellacrusca.it/it/consulenza/genere-dei-forestierismi/1229 about cases like this one. Is it more common `l'RFC/un RFC` or `la RFC/una RFC`?

Checked. I think RFC is properly used.

File: site/it/community/contribute/code.md (hunk 14 above)
Comment (codeadmin-peritiae) [reply site/it/community/contribute/code.md:64]: Comparing pros and cons, I prefer to use logging

File: site/it/community/contribute/code.md (hunk 16 above)
Comment (codeadmin-peritiae) [reply site/it/community/contribute/code.md:96]: Looking at the original English I prefer "Ci sarà un messaggio che evidenzia il branch". In my opinion it sounds better...

Comment (bhack) [reply site/it/community/contribute/code.md:96]: It Is also ok

File: site/it/community/contribute/code.md (hunk 2 above)
Comment (bhack) [reply site/it/community/contribute/code.md:3]: `Funzione obiettivo` It Is used but It Is not so dominant. I don't have stats but as we will find this quite frequently It Is better to collect other reviewers feedbacks here and choice If we want to put in the untrslated list or not.

File: site/it/community/contribute/code.md (hunk 7 above)
Comment (codeadmin-peritiae) [reply site/it/community/contribute/code.md:27]: A bug is a software defect so I don't see any reason to add an English word...

File: site/it/community/contribute/code.md (hunk 9 above)
Comment (codeadmin-peritiae) [reply site/it/community/contribute/code.md:37]: If you think is better, I agree.

File: site/it/community/contribute/code.md (hunk 11 above)
Comment (codeadmin-peritiae) [reply site/it/community/contribute/code.md:46]: About "bugs", please see my previous comment...

**Comment by bhack** [comment]
@mcaci Can you give a quick pass here? It was already reviewed.

**Comment by mcaci** [comment]
Yes, I have no additional feedbacks here, lgtm!
Thanks!

**Pull request merged**